modular_skyrat/modules/modular_items/lewd_items/code/lewd_arousal/status_effects/fluid_generation.dm

```diff
@@ -5,30 +5,30 @@
 	alert_type = null
 
 /datum/status_effect/body_fluid_regen/tick()
+	if(owner.stat >= DEAD || !owner.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
+		return
+
 	var/mob/living/carbon/human/affected_mob = owner
-	if(owner.stat != DEAD && affected_mob.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
-		var/obj/item/organ/external/genital/testicles/balls = owner.getorganslot(ORGAN_SLOT_TESTICLES)
-		var/obj/item/organ/external/genital/breasts/breasts = owner.getorganslot(ORGAN_SLOT_BREASTS)
-		var/obj/item/organ/external/genital/vagina/vagina = owner.getorganslot(ORGAN_SLOT_VAGINA)
+	var/obj/item/organ/external/genital/testicles/balls = owner.getorganslot(ORGAN_SLOT_TESTICLES)
+	var/obj/item/organ/external/genital/breasts/breasts = owner.getorganslot(ORGAN_SLOT_BREASTS)
+	var/obj/item/organ/external/genital/vagina/vagina = owner.getorganslot(ORGAN_SLOT_VAGINA)
 
-		var/interval = 5
-		if(balls)
-			if(affected_mob.arousal >= AROUSAL_LOW)
-				var/regen = (affected_mob.arousal / 25) * (balls.internal_fluids.maximum_volume / 235) * interval
-				balls.internal_fluids.add_reagent(/datum/reagent/consumable/cum, regen)
+	var/interval = 5
+	if(balls)
+		if(affected_mob.arousal >= AROUSAL_LOW)
+			var/regen = (affected_mob.arousal / 25) * (balls.internal_fluid_maximum / 235) * interval
+			balls.internal_fluid_count += regen
 
-		if(breasts)
-			if(breasts.lactates == TRUE)
-				var/regen = ((owner.nutrition / (NUTRITION_LEVEL_WELL_FED / 100)) / 100) * (breasts.internal_fluids.maximum_volume / 11000) * interval
-				if(!breasts.internal_fluids.holder_full())
-					owner.adjust_nutrition(-regen / 2)
-					breasts.internal_fluids.add_reagent(/datum/reagent/consumable/breast_milk, regen)
+	if(breasts)
+		if(breasts.lactates == TRUE)
+			var/regen = ((owner.nutrition / (NUTRITION_LEVEL_WELL_FED / 100)) / 100) * (breasts.internal_fluid_maximum / 11000) * interval
+			if(!breasts.internal_fluid_full())
+				owner.adjust_nutrition(-regen / 2)
+				breasts.adjust_internal_fluid(regen)
 
-		if(vagina)
-			if(affected_mob.arousal >= AROUSAL_LOW)
-				var/regen = (affected_mob.arousal / 25) * (vagina.internal_fluids.maximum_volume / 250) * interval
-				vagina.internal_fluids.add_reagent(/datum/reagent/consumable/femcum, regen)
-				if(vagina.internal_fluids.holder_full() && regen >= 0.15)
-					regen = regen
-			else
-				vagina.internal_fluids.remove_any(0.05)
+	if(vagina)
+		if(affected_mob.arousal >= AROUSAL_LOW)
+			var/regen = (affected_mob.arousal / 25) * (vagina.internal_fluid_maximum / 250) * interval
+			vagina.adjust_internal_fluid(regen)
+		else
+			vagina.adjust_internal_fluid(-0.05)
```

Comment:
Those numbers should all be defines, as they stay constant and don't change, so they actually have some sort of meaning for someone that arrives here later.